modules/gitops-argo-cd-round-robin-sharding-algorithm.adoc

```diff
@@ -0,0 +1,138 @@
+* Reduce the risk of bottlenecks
+* Improve overall performance and reliability of the Argo CD system
+
+The introduction of alternative sharding algorithms allows for further customization based on specific use cases. You can select the algorithm that best aligns with your deployment needs, which results in greater flexibility and adaptability in diverse operational scenarios.
+
+.Prerequisites
+* You have installed the {gitops-title} Operator in your cluster.
+* You have access to the {OCP} web console.
+* You have access to the cluster with `cluster-admin` privileges.
+
+.Procedure
+
+You can enable the `round-robin` sharding algorithm by either using the {OCP} web console or the command-line interface.
+
+* To enable the `round-robin` algorithm using the {OCP} web console, complete the following steps:
+
+. In the *Administrator* perspective of the web console, go to *Operators* → *Installed Operators*.
+
+. Click *{gitops-title}* from the installed operators and go to the *Argo CD* tab.
+
+. Click the Argo CD instance where you want to enable the `round-robin` sharding algorithm, for example `openshift-gitops`.
+
+. Click the *YAML* tab and edit the yaml file:
++
+.Example Argo CD instance with round-robin sharding algorithm enabled
+[source,yaml]
+----
+apiVersion: argoproj.io/v1beta1
+kind: ArgoCD
+metadata:
+  name: openshift-gitops
+  namespace: openshift-gitops
+spec:
+  controller:
+    processors: {}
+    sharding:
+      enabled: true <1>
+      replicas: 3 <2>
+    env: <3>
+      - name: ARGOCD_CONTROLLER_SHARDING_ALGORITHM 
+        value: round-robin
+----
+<1> Set the `sharding.enabled` parameter to `true` to enable sharding.
+<2> Set the number of replicas to the desired value, for example `3`.
+<3> Set the sharding algorithm to `round-robin`.
+
+. Click *Save* to save the changes.
+
+* To enable the `round-robin` algorithm using the command-line interface, complete the following steps:
+
+. Edit the Argo CD instance YAML file to enable sharding:
++
+[source,terminal]
+----
+$ oc patch argocd <argocd_instance> -n <namespace> --patch='{"spec":{"controller":{"sharding":{"enabled":true}}}}' --type=merge
+----
++
+.Example output
+[source,terminal]
+----
+argocd.argoproj.io/<argocd_instance> patched
+----
+
+. Configure the sharding algorithm to `round-robin`:
++
+[source,terminal]
+----
+$ oc patch argocd <argocd_instance> -n <namespace> --patch='{"spec":{"controller":{"env":[{"name":"ARGOCD_CONTROLLER_SHARDING_ALGORITHM","value":"round-robin"}]}}}' --type=merge
+----
++
+.Example output
+[source,terminal]
+----
+argocd.argoproj.io/<argocd_instance> patched
+----
+
+. Set the number of replicas to the desired value:
++
+[source,terminal]
+----
+oc patch argocd <argocd_instance> -n <namespace> --patch='{"spec":{"controller":{"sharding":{"replicas":<value>}}}}' --type=merge
+----
++
+.Example output
+[source,terminal]
+----
+argocd.argoproj.io/<argocd_instance> patched
+----
+
+.Verification
+
```

Comment:
@ekristova Can we add this statement from the SME draft here: "The argocd-application-controller pods should be rolled out one by one and you should be able to see the logs on each shard"?

Is there a possibility to capture this in the UI as well? Please discuss with SME about this.

Reply:
Hello @Srivaralakshmi The statement applies to the command that would be similar to this:
`oc logs --prefix  -l app.kubernetes.io/name=openshift-gitops-application-controller -n openshift-gitops`

But in my opinion, the output is too long and hard to read:
![image](https://github.com/openshift/openshift-docs/assets/98941675/347b085d-bf72-4e38-9008-aff9b39c0d8e)

For that reason, I decided to include an example that just shows logs for one specific pod:
![image](https://github.com/openshift/openshift-docs/assets/98941675/1f91316f-2ce2-40d0-97bb-5af5abd74a94)

But to make it clearer, I think it is a good idea to include a callout (1) "This example shows the logs for shard 1." , so that it would look similar to this:
![image](https://github.com/openshift/openshift-docs/assets/98941675/d577c037-f45d-4b05-92e5-3238a0149bc2)

WDYT?

Reply:
Oh! I just realized, the statement itself is true even for the shorter command right? So I can add it anyway! Sorry for the misunderstanding!

Reply:
Or... since I am now including the steps to verify the number of pods and to view the logs, the sentence is actually unnecessary as it explains the same thing. WDYT?